src/error.rs

```diff
@@ -17,15 +17,45 @@ use core::num::NonZeroU32;
 /// Internally this type is a NonZeroU32, with certain values reserved for
 /// certain purposes, see [`Error::INTERNAL_START`] and [`Error::CUSTOM_START`].
 #[derive(Copy, Clone, Eq, PartialEq)]
-pub struct Error(NonZeroU32);
+pub struct Error(pub(crate) NonZeroU32);
+
+// TODO: Convert to a function when min_version >= 1.33
+macro_rules! internal_error {
+    ($n:expr) => {
+        Error(unsafe { NonZeroU32::new_unchecked(Error::INTERNAL_START + $n as u16 as u32) })
+    };
+}
 
 impl Error {
+    /// This target/platform is not supported by `getrandom`.
+    pub const UNSUPPORTED: Error = internal_error!(0);
+    /// The platform-specific `errno` returned a non-positive value.
+    pub const ERRNO_NOT_POSITIVE: Error = internal_error!(1);
+    /// Invalid conversion from a non-standard [`std::io::Error`]
+    pub const UNKNOWN_IO_ERROR: Error = internal_error!(2);
+    /// Call to [`SecRandomCopyBytes`](https://developer.apple.com/documentation/security/1399291-secrandomcopybytes) failed.
+    pub const SEC_RANDOM_FAILED: Error = internal_error!(3);
+    /// Call to [`RtlGenRandom`](https://docs.microsoft.com/en-us/windows/win32/api/ntsecapi/nf-ntsecapi-rtlgenrandom) failed.
+    pub const RTL_GEN_RANDOM_FAILED: Error = internal_error!(4);
+    /// RDRAND instruction failed due to a hardware issue.
+    pub const FAILED_RDRAND: Error = internal_error!(5);
+    /// RDRAND instruction unsupported on this target.
+    pub const NO_RDRAND: Error = internal_error!(6);
+    /// Using `wasm-bindgen`, browser does not support `self.crypto`.
+    pub const BINDGEN_CRYPTO_UNDEF: Error = internal_error!(7);
+    /// Using `wasm-bindgen`, browser does not support `crypto.getRandomValues`.
+    pub const BINDGEN_GRV_UNDEF: Error = internal_error!(8);
+    /// Using `stdweb`, no cryptographic RNG is available.
+    pub const STDWEB_NO_RNG: Error = internal_error!(9);
+    /// Using `stdweb`, invoking a cryptographic RNG failed.
+    pub const STDWEB_RNG_FAILED: Error = internal_error!(10);
+
     #[deprecated(since = "0.1.7")]
     /// Unknown error.
-    pub const UNKNOWN: Error = UNSUPPORTED;
+    pub const UNKNOWN: Error = Error::UNSUPPORTED;
     #[deprecated(since = "0.1.7")]
     /// System entropy source is unavailable.
-    pub const UNAVAILABLE: Error = UNSUPPORTED;
+    pub const UNAVAILABLE: Error = Error::UNSUPPORTED;
```

Comment:
I think we can remove these two deprecated constants as part of this PR.

Reply:
Done